### ISSUE TITLE
Crashing in Kubernetes

### ISSUE THREAD

**Issue by tesharp** [Azure/AppConfiguration#848]
Tried to set up the AzureAppConfigurationProvider in AKS using the 1.0.0 release. After creating the first configuration the provider just starts crashing.

Not been able to sync any configurations so far. There are key vault references in the configuration

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x153cdfb]

goroutine 123 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x16a2500, 0x2651990})
        /usr/local/go/src/runtime/panic.go:884 +0x213
azappconfig/provider/internal/providers.(*ConfigurationSettingProvider).CreateKeyValueSettings(0xc0002118c0, {0x1aed2e8?, 0xc0008d7ec0}, {0x0, 0x0})
        /workspace/internal/providers/configuration_setting_provider.go:138 +0xa3b
azappconfig/provider/internal/controllers.(*AppConfigurationProviderProcessor).ProcessFullReconciliation(0xc000459a38)
        /workspace/internal/controllers/processor.go:60 +0x52
azappconfig/provider/internal/controllers.(*AppConfigurationProviderProcessor).PopulateSettings(0x168fe00?, 0xc00047c0c0?)
        /workspace/internal/controllers/processor.go:43 +0x25
azappconfig/provider/internal/controllers.(*AzureAppConfigurationProviderReconciler).Reconcile(0xc00047c0f0, {0x1aed2e8, 0xc0008d7ec0}, {{{0xc00050f100, 0x10}, {0xc000aaa180, 0x22}}})
        /workspace/internal/controllers/appconfigurationprovider_controller.go:177 +0xe4e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1aed2e8?, {0x1aed2e8?, 0xc0008d7ec0?}, {{{0xc00050f100?, 0x16272c0?}, {0xc000aaa180?, 0x40f926?}}})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002683c0, {0x1aed240, 0xc0003f5f90}, {0x16fdc20?, 0xc0008b49a0?})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:323 +0x377
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002683c0, {0x1aed240, 0xc0003f5f90})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:231 +0x587
```

**Comment by linglingye001** [comment]
Thank you for bringing this issue to our attention. Could you please provide us with configuration of `AzureAppConfigurationProvider` and any other relevant information? It will help us better understand the problem and work towards a resolution. Thanks!

**Comment by jimmyca15** [comment]
To add a bit to @linglingye001 has mentioned. This error occurs whenever a key-value is pulled from App Configuration that has a `null` value. Until the fix that was mentioned is provided, this error can be avoided by ensuring that no key-values that are referenced in App Configuration have a `null` value.

**Comment by RichardChen820** [comment]
@tesharp  Fixed in v1.1.0
